include/nuttx/sched_note.h

```diff
@@ -255,16 +274,28 @@ struct note_syscall_leave_s
 
 struct note_irqhandler_s
 {
-  struct note_common_s nih_cmn; /* Common note parameters */
-  int nih_irq;                  /* IRQ number */
+  struct note_common_s nih_cmn;        /* Common note parameters */
+  uint8_t nih_irq;                     /* IRQ number */
+  uint8_t nih_handler[sizeof(void *)]; /* Address of the interrupt handler */
```

Comment:
It's better to move nih_handler related change to another patch, so we can merge it to mainline directly. Also please don't forget to update the apps/system/sched_note/.